pkg/monitoring/monitor.go

```diff
@@ -168,21 +167,37 @@ func (m Monitor) Run() {
 			NewInstrumentedSourceFactory(factory, m.ChainMetrics))
 	}
 
+	// setup per-feed monitor
 	monitor := NewMultiFeedMonitor(
 		m.ChainConfig,
 		m.Log,
 		instrumentedSourceFactories,
 		m.ExporterFactories,
 		100, // bufferCapacity for source pollers
 	)
-
 	subs.Go(func() {
 		m.Manager.Run(rootCtx, func(localCtx context.Context, data RDDData) {
 			m.ChainMetrics.SetNewFeedConfigsDetected(float64(len(data.Feeds)))
 			monitor.Run(localCtx, data)
 		})
```

Comment:
Run() doesn't work as you expect - ie. it does NOT "broadcast" rdd data to all the callbacks registered with Run. It's designed to only be run once! I suggest an easy alternative:

```suggestion
		m.Manager.Run(rootCtx, func(localCtx context.Context, data RDDData) {
			m.ChainMetrics.SetNewFeedConfigsDetected(float64(len(data.Feeds)))
			monitor.Run(localCtx, data)
			networkMonitor.Run(localCtx, data)
		})
```